html/js/tree/Node.js

```diff
@@ -37,38 +38,105 @@ GTE.TREE = (function (parentModule) {
             this.reachedBy.draw();
         }
         var thisNode = this;
-        var circle = GTE.canvas.circle(GTE.CONSTANTS.CIRCLE_SIZE)
-            .addClass('node')
-            .x(this.x)
-            .y(this.y)
-            .click(function() {
-                thisNode.onClick();
-            });
+        if (this.player && this.player.id === 0){
+            this.shape = GTE.canvas.rect(
+                          GTE.CONSTANTS.CIRCLE_SIZE, GTE.CONSTANTS.CIRCLE_SIZE);
+        } else {
+            this.shape = GTE.canvas.circle(GTE.CONSTANTS.CIRCLE_SIZE);
+        }
+        this.shape.addClass('node')
+                  .x(this.x)
+                  .y(this.y)
+                  .click(function() {
+                      thisNode.onClick();
+                  });
+        if (this.player) {
+            this.shape.fill(this.player.colour);
+            this.drawPlayer();
+        } else {
+            this.shape.fill(GTE.COLOURS.BLACK);
+        }
+
+        if (GTE.MODE === GTE.MODES.PLAYERS && this.isLeaf()) {
+            this.shape.hide();
+        }
+    };
+
+    /**
+    * Draws the player. It needs to be done within the Node so that there is
+    * an instance of ContentEditable per Node
+    */
+    Node.prototype.drawPlayer = function () {
+        var thisPlayer = this.player;
+        this.playerNameText = thisPlayer.draw(
+            this.x + GTE.CONSTANTS.TEXT_NODE_MARGIN, this.y);
+        if (this.player.id === 0 && !GTE.tree.showChanceName) {
+            this.playerNameText.hide();
+        }
+    };
+
+    /**
+    * Toggles the visibility of the default name text
+    */
+    Node.prototype.togglePlayerName = function () {
```

Comment:
Can we rename this to "togglePlayerNameVisibility"? I think jQuery has a function that toggles visibility called just "toggle", but I think that with the PlayerName appendix it gets too ambiguous.